### PR TITLE
[css-transforms] scale function and property %

### LINKED ISSUE #3399: [css-transforms] Have scale function accept percentage value
Hello,
I would like to suggest adding percentage for [scale](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/scale) function in transformations.

Current state: `scale()` is using `[-NUM, NUM]` where NUM is whole number (integer)
Suggestion:` scale()` will support `[-NUM, NUM]` and `[-%, %]`
Example: 
`transform: scale(120%)` will work same as `transform:scale(1.2)` is working right now.

It is not a game-breaker but it is nice to have.

Thank you for your work.

### PR DESCRIPTION
**Pull request by ewilligers**
Allow percentages inside the scale functions,
and the scale property.

resolves #3399

### CODE REVIEW

**Comment by smfr** [comment]
This change should be made in css-transforms-2, not css-transforms-1.

**Comment by ewilligers** [comment]
I have updated to PR to only involve css-transforms-2. I have also described serialization.

**Comment by ewilligers** [comment]
@tabatkins @smfr @grorg @hober please review

File: css-transforms-2/Overview.bs [hunk 1]
```diff
@@ -627,6 +627,10 @@ If three values are given,
 this specifies a 3d scaling,
 equivalent to the ''scale3d()'' function.
 
+A <<percentage>> is equivalent to a <<number>>,
+for example ''scale: 100%;'' is equivalent to ''scale: 1;''.
+Numbers are used during serialization.
```
Comment (smfr) [inline css-transforms-2/Overview.bs]: I think this should move to section 4.1, about serialization.

File: css-transforms-2/Overview.bs [hunk 2]
```diff
@@ -878,6 +882,19 @@ The Transform Functions {#transform-functions}
 =======================
 
 The value of the 'transform' property is a list of <dfn>&lt;transform-function></dfn>. The set of allowed transform functions is given below. Wherever <<angle>> is used in this specification, a <<number>> that is equal to zero is also allowed, which is treated the same as an angle of zero degrees. A percentage for horizontal translations is relative to the width of the <a>reference box</a>. A percentage for vertical translations is relative to the height of the <a>reference box</a>.
+A percentage in a scale function is equivalent to a number, and serializes as a number.
```
Comment (smfr) [inline css-transforms-2/Overview.bs]: I think this should move to section 4.1, about serialization.

**Comment by smfr** [comment]
Ah it's section 2.1 in transforms-2

**Comment by ewilligers** [comment]
Section 2.1 is about computed value serialization. Percentages in scale serialize as numbers even in the specified value. Are you requesting a new section about specified value serialization?

**Comment by smfr** [comment]
It wasn't clear from "serializes as a number" whether you were talking about specified or computed. Maybe clarify?

**Comment by ewilligers** [comment]
> It wasn't clear from "serializes as a number" whether you were talking about specified or computed. Maybe clarify?

Updated.

**Review by alshamiri5 — APPROVED** [review]

**Comment by ewilligers** [comment]
Waiting for approval from an editor.  `css-transforms-2/Overview.bs` lists

- Tab Atkins Jr.
- Simon Fraser
- Dean Jackson
- Theresa O'Connor

**Review by smfr — APPROVED** [review]

**Pull request merged**